src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,151 @@
+
+    if ComputeFleetStatus.is_stop_requested(computefleet_status):
+        _stop_partition(config)
+    elif ComputeFleetStatus.is_start_requested(computefleet_status):
+        _start_partition()
+
+
+def _start_partition():
+    log.info("Setting slurm partition to UP and resuming nodes...")
+    update_all_partitions(PartitionStatus.UP, reset_node_addrs_hostname=False)
+    resume_powering_down_nodes()
+
+
+def _stop_partition(config):
+    log.info("Setting slurm partition to INACTIVE and terminating all compute nodes...")
+    update_all_partitions(PartitionStatus.INACTIVE, reset_node_addrs_hostname=True)
+    instance_manager = InstanceManager(
+        config.region,
+        config.cluster_name,
+        config.boto3_config,
+    )
+    instance_manager.terminate_all_compute_nodes(config.terminate_max_batch_size)
+
+
+def _get_computefleet_status(computefleet_status_data_path):
+    try:
+        with open(computefleet_status_data_path, "r", encoding="utf-8") as computefleet_status_data_file:
+            computefleet_status = ComputeFleetStatus(
+                json.load(computefleet_status_data_file).get(ComputeFleetStatusManager.COMPUTE_FLEET_STATUS_ATTRIBUTE)
+            )
+        log.info("ComputeFleet status is: %s", computefleet_status)
+    except IOError:
+        log.error(f"Cannot read ComputeFleet status data file: {computefleet_status_data_path}")
+        raise
+
+    return computefleet_status
+
+
+def main():
+    default_log_file = "/var/log/parallelcluster/slurm_fleet_manager.log"
+    logging.basicConfig(
+        filename=default_log_file,
+        level=logging.INFO,
+        format="%(asctime)s - [%(name)s:%(funcName)s] - %(levelname)s - %(message)s",
+    )
+    log.info("FleetManager startup.")
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-cf", "--computefleet-status-data", help="Path to compute fleet status data", required=True)
+    args = parser.parse_args()
+    try:
+        config_file = os.environ.get(
+            "CONFIG_FILE", os.path.join(CONFIG_FILE_DIR, "parallelcluster_slurm_fleet_manager.conf")
+        )
+        fleet_manager_config = SlurmFleetManagerConfig(config_file)
+        try:
+            # Configure root logger
+            fileConfig(fleet_manager_config.logging_config, disable_existing_loggers=False)
+        except Exception as e:
+            log.warning(
+                "Unable to configure logging from %s, using default settings and writing to %s.\nException: %s",
+                fleet_manager_config.logging_config,
+                default_log_file,
+                e,
+            )
+        log.info("FleetManager config: %s", fleet_manager_config)
+        _fleet_manager(fleet_manager_config, args.computefleet_status_data)
+        log.info("FleetManager finished.")
+    except Exception as e:
+        log.exception("Encountered exception when running fleet manager: %s", e)
```

Comment:
I'm afraid the program is not exiting with an error code in case of exception

Reply:
added system exit 1